slasher/detection/detect.go

```diff
@@ -216,3 +216,37 @@ func isSurrounding(incomingAtt *ethpb.IndexedAttestation, prevAtt *ethpb.Indexed
 	return incomingAtt.Data.Source.Epoch < prevAtt.Data.Source.Epoch &&
 		incomingAtt.Data.Target.Epoch > prevAtt.Data.Target.Epoch
 }
+
+
+
+
+
+
+
+func (ds *Service) UpdateHighestAttestation(ctx context.Context, att *ethpb.IndexedAttestation) error {
+	for _, idx := range att.AttestingIndices {
+		h, err := ds.slasherDB.HighestAttestation(ctx, idx)
+		if err != nil {
+			return err
+		}
+		update := false
+		if h.HighestSourceEpoch < att.Data.Source.Epoch {
+			h.HighestSourceEpoch = att.Data.Source.Epoch
+			update = true
+		}
+		if h.HighestTargetEpoch < att.Data.Target.Epoch {
+			h.HighestTargetEpoch = att.Data.Target.Epoch
+			update = true
+		}
+
+		if update {
+			if err := ds.slasherDB.SaveHighestAttestation(ctx, idx, h); err != nil {
```

Comment:
what if highest target is higher but source is lower . do we want to update highest source to a lower value? (same for the opposite)

Reply:
can that even happen without slashing? have an example?

Reply:
@shayzluf I think in that case it's a slashable attestation

Reply:
it is slashable but i think slasher protection should still protect it on the best way (might not be caught as slashingand will be slasher by the later offence)